docs/guides/data/mounting_data.md

```diff
@@ -1,105 +1,186 @@
-In this guide, we will explain how to attach a datasource to your job.
+In this guide, we will explain how to attach one or more datasets to a job.
+First, let's review some basics about FloydHub datasets.
+
+### Datasets
+
+A Floyd dataset is a directory (folder) of data that can be used during a
+job. To create a new dataset, please follow
+[this guide](../basics/create_new/#create-a-new-dataset). You can view the
+datasets you have created in the datasets page in the dashboard. You can also
+view public datasets by searching for them on FloydHub.
+
+#### Why keep data separate from code?
+
+A data scientist tweaks his/her code often during the process of creating a
+deep-learning model. However, he/she doesn't change the underlying data nearly
+as often, if at all.
+
+Each time you run a job on FloydHub, a copy of your code is uploaded to
+FloydHub and run on one of FloyHub's powerful deep-learning servers. Because
+your data isn't changing from job to job, it wouldn't make sense to keep your
+data with your code and upload it with each job. Instead, we upload a
+dataset once, and attach, or "mount", it to each job. This saves a
+significant amount of time on each job.
```

Comment:
Another benefit would be sharing data for easy collaboration with others ;)

Reply:
Good call. I'll add it

Reply:
Done :slightly_smiling_face: